packages/env/lib/config/parse-config.js

```diff
@@ -3,206 +3,421 @@
- * @return {WPServiceConfig} Parsed environment-level configuration.
+ * @param {string} options.shouldInferType    Indicates whether or not we should infer the type of project wp-env is being used in.
+ * @param {string} options.cacheDirectoryPath Path to the work directory located in ~/.wp-env.
+ *
+ * @return {Promise<WPServiceConfig>} The default config object.
  */
-module.exports = async function parseConfig( config, options ) {
-	return {
-		port: config.port,
-		phpVersion: config.phpVersion,
-		coreSource: includeTestsPath(
-			await parseCoreSource( config.core, options ),
-			options
-		),
-		pluginSources: config.plugins.map( ( sourceString ) =>
-			parseSourceString( sourceString, options )
-		),
-		themeSources: config.themes.map( ( sourceString ) =>
-			parseSourceString( sourceString, options )
-		),
-		config: config.config,
-		mappings: Object.entries( config.mappings ).reduce(
-			( result, [ wpDir, localDir ] ) => {
-				const source = parseSourceString( localDir, options );
-				result[ wpDir ] = source;
-				return result;
+async function getDefaultConfig(
+	configDirectoryPath,
+	{ shouldInferType, cacheDirectoryPath }
+) {
+	// Our default config should try to infer what type of project
+	// this is in order to automatically map the current directory.
+	const type = shouldInferType
+		? await detectDirectoryType( configDirectoryPath )
+		: null;
+
+	const rawConfig = {
+		core: type === 'core' ? '.' : null,
+		phpVersion: null,
+		plugins: type === 'plugin' ? [ '.' ] : [],
+		themes: type === 'theme' ? [ '.' ] : [],
+		port: 8888,
+		mappings: {},
+		config: {
+			WP_DEBUG: true,
+			SCRIPT_DEBUG: true,
+			WP_ENVIRONMENT_TYPE: 'local',
+			WP_PHP_BINARY: 'php',
+			WP_TESTS_EMAIL: 'admin@example.org',
+			WP_TESTS_TITLE: 'Test Blog',
+			WP_TESTS_DOMAIN: 'localhost',
+			WP_SITEURL: 'http://localhost',
+			WP_HOME: 'http://localhost',
+		},
+		env: {
+			development: {}, // No overrides needed, but it should exist.
+			tests: {
+				config: { WP_DEBUG: false, SCRIPT_DEBUG: false },
+				port: 8889,
 			},
-			{}
-		),
+		},
 	};
-};
 
-async function parseCoreSource( coreSource, options ) {
-	// An empty source means we should use the latest version of WordPress.
-	if ( ! coreSource ) {
-		const wpVersion = await getLatestWordPressVersion();
-		if ( ! wpVersion ) {
-			throw new ValidationError(
-				'Could not find the latest WordPress version. There may be a network issue.'
-			);
-		}
+	return await parseRootConfig( 'default', rawConfig, {
+		cacheDirectoryPath,
+	} );
+}
 
-		coreSource = `WordPress/WordPress#${ wpVersion }`;
+/**
+ * Gets a service configuration object containing overrides from our environment variables.
+ *
+ * @param {string} cacheDirectoryPath Path to the work directory located in ~/.wp-env.
+ *
+ * @return {WPServiceConfig} An object containing the environment variable overrides.
+ */
+function getEnvironmentVarOverrides( cacheDirectoryPath ) {
```

Comment:
I noticed that this config object works differently from the other pieces, and I'm wondering if it should also go through the parser/validator? For example, I notice in `getConfigFromEnvironmentVars` that we do the source parsing when reading the environment variables, but for the other configs, we do source parsing during `parseServiceConfig`. 

I wonder if it'd be simpler to just have a single function for this (instead of also having `getConfigFromEnvironmentVars`), and then just run that more barebones config object through `parseRootConfig` (or `parseServiceConfig`, whichever makes sense) and let that handle the extra steps

Reply:
This was mostly just a consequence of copy/pasting the original environment variable parsing with minimal changes. One consideration though is that, as-is, we get the nice presentation of error messages from the validation done there. If we used `parseServiceConfig()` we would get the `configFile` and `development.port` style error messages.

I think that, although it's different, I would rather keep it the way it is. It seems better to perform the parsing on environment variables in a very granular way like this instead of trying to pass it through the other parsing logic.

Reply:
Fair enough 👍